src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/Designer/DataTransferTypes.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Debug.Designer
+{
+    internal sealed record PropertyValueChangeRequest(
+        string ProfileName,
+        string PropertyName,
+        object Value);
+
+    internal sealed record Profile(string Name, string DisplayName);
+
+    internal sealed record Category(string Name, string DisplayName);
+
+    internal sealed record PropertyEditor(string Name, ImmutableDictionary<string, string> Metadata);
+
+    internal sealed record PropertyMetadata(
+        string Name,
+        string DisplayName,
+        Profile Profile,
+        Category Category,
+        string? DependsOn,
+        string? Description,
+        int Priority,
+        PropertyEditor? Editor,
+        string? SearchTerms,
+        string? HelpUrl,
+        string? VisibilityCondition);
+
+    internal sealed record SupportedValue(
+        string DisplayName,
+        object Value);
+
+    internal sealed record Property(
```

Comment:
Note that these DTOs are property-centric. Instead of a multi-level hierarchy where we start with a list of profiles and from there can dig down to the properties on the individual page, we have a flat list of properties with enough metadata to determine which pages they belong to.

Reply:
I've commented elsewhere that I'm not sure about this approach. Conceptually, each profile is an isolated set of properties, unlike in the case of project properties where there are relationships between each. I describe elsewhere how some changing this can make things like notification of new profiles more straightforward.

Reply:
I've now updated this to be profile-centric rather than property-centric. The client now receives profiles one at time and each profile contains its properties, rather than the client receiving all properties at once and using property metadata to divide them into pages.